web/client/plugins/MetadataExplorer.jsx

```diff
@@ -0,0 +1,189 @@
+/**
+ * Copyright 2016, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+const React = require('react');
+const {connect} = require('react-redux');
+const assign = require('object-assign');
+const {cswToCatalogSelector} = require("../selectors/cswtocatalog");
+// const {createDefaultMemorizeSelector} = require("../selectors/common");
+const {createSelector} = require("reselect");
+const {Glyphicon, Input, Alert, Pagination} = require('react-bootstrap');
+const Spinner = require('react-spinkit');
+const {textSearch} = require("../actions/catalog");
+const {addLayer} = require("../actions/layers");
+const {zoomToExtent} = require("../actions/map");
+const Message = require("../components/I18N/Message");
+require('./metadataexplorer/css/style.css');
+const makeCatalogSelector = () => {
+    return createSelector([(state) => state.catalog], cswToCatalogSelector);
+};
+
+const makeMapStateToProps = () => {
+    const getCatalogRecords = makeCatalogSelector();
+    const mapStateToProps = (state, props) => {
+        return {
+          records: getCatalogRecords(state, props)
+      };
+    };
+    return mapStateToProps;
+};
+
+const RecordGrid = connect(makeMapStateToProps, {
+    // add layer action to pass to the layers
+    onZoomToExtent: zoomToExtent
+})(require("../components/catalog/RecordGrid"));
+
+
+const MetadataExplorerComponent = React.createClass({
```

Comment:
Can not do it for the moment, because RecordGrid is connected, and can not be passed as a property as it is.